src/components/Products/ProductList.jsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { getAllProducts, deleteProduct } from "../../api/productService";
+import ProductForm from "./ProductForm";
+import { BASE_URL } from "../../api/axiosInstance";
+
+const ProductList = ({ setBasket }) => {
+  const [products, setProducts] = useState([]);
+  const [userRole, setUserRole] = useState("");
+
+  useEffect(() => {
+    loadProducts();
+    const token = localStorage.getItem("token");
+    const userRole = localStorage.getItem("user-role");
+    if (token) {
+      setUserRole(userRole);
+    }
+  }, []);
+
+  const loadProducts = async () => {
+    const products = await getAllProducts();
+    setProducts(products);
+  };
+
+  const handleDelete = async (id) => {
+    await deleteProduct(id);
+    loadProducts();
+  };
+
+  const addToBasket = (product) => {
+    setBasket((prev) => [...prev, product]);
+  };
+
+  return (
+    <div>
+      <h1 className="text-3xl font-bold mb-4">Products</h1>
+      {userRole === "admin" && <ProductForm onProductAdded={loadProducts} />}
```

Comment:
we should keep ProductForm in pages/Products level and keeping this component only for products list